controllers/sriovnetworknodepolicy_controller.go

```diff
@@ -287,13 +290,17 @@ func (r *SriovNetworkNodePolicyReconciler) syncSriovNetworkNodeState(np *sriovne
 				// Merging only for policies with the same priority (ppp == p.Spec.Priority)
 				// This boolean flag controls merging of PF configuration (e.g. mtu, numvfs etc)
 				// when VF partition is configured.
-				p.Apply(found, bool(ppp == p.Spec.Priority))
+				p.Apply(newVersion, bool(ppp == p.Spec.Priority))
 				// record the evaluated policy priority for next loop
 				ppp = p.Spec.Priority
 			}
 		}
-		found.Spec.DpConfigVersion = cksum
-		err = r.Update(context.TODO(), found)
+		newVersion.Spec.DpConfigVersion = cksum
+		if reflect.DeepEqual(found.Spec, newVersion.Spec) {
```

Comment:
Shall we also user `equality.Semantic.DeepDerivative` here?

Reply:
I added it in the daemonset as `deepequal` wasn't working, but I guess it won't hurt / will make the code future proof.
Will change

Reply:
Done